File: leetcode/easy/0530. Minimum Absolute Difference in BST/main.go
package main

import "math"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func getMinimumDifference(root *TreeNode) int {
	result := math.MaxInt

	var (
		inorder func(root *TreeNode)
		prev    *int
	)
	inorder = func(root *TreeNode) {
		if root == nil {
			return
		}

		inorder(root.Left)

		if prev != nil {
			diff := root.Val - *prev
			if diff < result {
				result = diff
			}
		}

		val := root.Val
		prev = &val

		inorder(root.Right)
	}

	inorder(root)

	return result
}

File: leetcode/easy/0530. Minimum Absolute Difference in BST/main.ts
// Solved by @kitanoyoru

class TreeNode {
  val: number
  left: TreeNode | null
  right: TreeNode | null
  constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
    this.val = val === undefined ? 0 : val
    this.left = left === undefined ? null : left
    this.right = right === undefined ? null : right
  }
}

const getMinimumDifference = (root: TreeNode | null): number => {
  let ans = Number.MAX_SAFE_INTEGER,
    prev = -1

  const dfs = (node: TreeNode | null) => {
    if (node == null) {
      return
    }
    dfs(node.left)
    if (prev != -1) {
      ans = Math.min(ans, Math.abs(node.val - prev))
    }
    prev = node.val
    dfs(node.right)
  }

  dfs(root)

  return ans
}

export {}
